src/log/log.h

```diff
@@ -125,6 +125,50 @@ struct __wt_txn_printlog_args {
     uint32_t flags;
 };
 
+struct __wt_log_manager {
+
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+#define WT_LOG_CONFIG_ENABLED 0x001u  /* Logging is configured */
+#define WT_LOG_DOWNGRADED 0x002u      /* Running older version */
+#define WT_LOG_ENABLED 0x004u         /* Logging is enabled */
+#define WT_LOG_EXISTED 0x008u         /* Log files found */
+#define WT_LOG_FORCE_DOWNGRADE 0x010u /* Force downgrade */
+#define WT_LOG_INCR_BACKUP 0x020u     /* Incremental backup log required */
+#define WT_LOG_RECOVER_DIRTY 0x040u   /* Recovering unclean */
+#define WT_LOG_RECOVER_DONE 0x080u    /* Recovery completed */
+#define WT_LOG_RECOVER_ERR 0x100u     /* Error if recovery required */
+#define WT_LOG_RECOVER_FAILED 0x200u  /* Recovery failed */
+#define WT_LOG_REMOVE 0x400u          /* Removal is enabled */
+#define WT_LOG_ZERO_FILL 0x800u       /* Manually zero files */
+                                      /* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
+    uint32_t flags;                   /* Global logging configuration */
+    WT_CONDVAR *cond;                 /* Log server wait mutex */
+    WT_SESSION_IMPL *session;         /* Log server session */
+    wt_thread_t tid;                  /* Log server thread */
+    bool tid_set;                     /* Log server thread set */
+    WT_CONDVAR *file_cond;            /* Log file thread wait mutex */
+    WT_SESSION_IMPL *file_session;    /* Log file thread session */
+    wt_thread_t file_tid;             /* Log file thread */
+    bool file_tid_set;                /* Log file thread set */
+    WT_CONDVAR *wrlsn_cond;           /* Log write lsn thread wait mutex */
+    WT_SESSION_IMPL *wrlsn_session;   /* Log write lsn thread session */
+    wt_thread_t wrlsn_tid;            /* Log write lsn thread */
+    bool wrlsn_tid_set;               /* Log write lsn thread set */
+    WT_LOG *log;                      /* Logging structure */
+    WT_COMPRESSOR *compressor;        /* Logging compressor */
+    wt_shared uint32_t cursors;       /* Log cursor count */
+    wt_off_t dirty_max;               /* Log dirty system cache max size */
+    wt_off_t file_max;                /* Log file max size */
+    wt_off_t extend_len;              /* file_extend log length */
+    uint32_t force_write_wait;        /* Log force write wait configuration */
+    const char *log_path;             /* Logging path format */
+    uint32_t prealloc;                /* Log file pre-allocation */
+    uint32_t prealloc_init_count;     /* initial number of pre-allocated log files */
+    uint16_t req_max;                 /* Max required log version */
+    uint16_t req_min;                 /* Min required log version */
```

Comment:
I'm also happy to do this in a separate ticket if it's easier.
We discussed offline about most of these fields being private. Can we either tag them with the `#private` comment or break into private substructs?

Reply:
I have added the "private" tag to the members. 
I tried to move the private members to a separate struct, then I realised that it creates a dependency between log.h & log_private.h
As of now log.h doesn't include log_private.h, I prefer to have it this way. 
If we create a dependency of log.h on log_private.h and include it, some other module including "log.h" and accessing a private structure part of "log_private.h" will not be detected by compiler.

Reply:
I think we've already lost those compiler protections by #including `log_private.h` from `log_inline.h`, so I don't think it's a big issue.

I'm happy to stick with individual tags, but for future reference we should also be able to create private sub-structs without adding the dependency. 

`s_prototypes` creates the 
```
struct __wt_log_thread;
typedef struct __wt_log_thread WT_LOG_THREAD;
```
forward declarations in `wt_internal.h`. So as long as we declare the sub-structs as pointers
```
struct __wt_log_manager {
    WT_LOG_THREAD *tid
};
```
and only de-reference inside .c files which `#include log_private.h` we'll be ok.